modules/websocket_streaming_client_module/src/websocket_streaming_client_module_impl.cpp

```diff
@@ -15,35 +15,42 @@ static const char* WebsocketDevicePrefix = "daq.lt://";
 using namespace discovery;
 using namespace daq::websocket_streaming;
 
+static std::vector<ServerCapabilityCb> getServerCapabilitiesCb()
+{
+    return {
+        [](MdnsDiscoveredDevice discoveredDevice)
+        {
+            auto connectionString = fmt::format("{}{}:{}{}",
+                                WebsocketDevicePrefix,
+                                discoveredDevice.ipv4Address,
+                                discoveredDevice.servicePort,
+                                discoveredDevice.getPropertyOrDefault("path", "/"));
+            auto cap = ServerCapability("opendaq_lt_streaming", "openDAQ LT Streaming", ProtocolType::Streaming).addConnectionString(connectionString).setConnectionType("Ipv4");
+            cap.setPrefix("daq.lt");
+            return cap;
+        }
+    };
+}
+
 WebsocketStreamingClientModule::WebsocketStreamingClientModule(ContextPtr context)
     : Module("openDAQ websocket client module",
             daq::VersionInfo(WS_STREAM_CL_MODULE_MAJOR_VERSION, WS_STREAM_CL_MODULE_MINOR_VERSION, WS_STREAM_CL_MODULE_PATCH_VERSION),
             std::move(context),
             "WebsocketStreamingClient")
     , deviceIndex(0)
-    , discoveryClient(
-        {
-            [context = this->context](MdnsDiscoveredDevice discoveredDevice)
-            {
-                auto connectionString = fmt::format("{}{}:{}{}",
-                                   WebsocketDevicePrefix,
-                                   discoveredDevice.ipv4Address,
-                                   discoveredDevice.servicePort,
-                                   discoveredDevice.getPropertyOrDefault("path", "/"));
-                auto cap = ServerCapability("opendaq_lt_streaming", "openDAQ LT Streaming", ProtocolType::Streaming).addConnectionString(connectionString).setConnectionType("Ipv4");
-                cap.setPrefix("daq.lt");
-                return cap;
-            }
-        },
-        {"WS"}
-    )
+    , discoveryClient(getServerCapabilitiesCb(), {"LT"})
+    , oldDiscoveryClient(getServerCapabilitiesCb(), {"WS"})
```

Comment:
The discovery client can be modified to accept several service names, just multiple queries should be created. The caps might need to be refactored a bit.